workshop/session_3/09_biopython_sequences.ipynb

```diff
@@ -0,0 +1,856 @@
+    "print(align_info.pos_specific_score_matrix())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "editable": false
+   },
+   "source": [
+    "## Downloading and reading FASTA files\n",
+    "\n",
+    "FASTA is a simple, plain-text-based sequence format for representing sequences, which supports multiple records contained in the same file. An example:\n",
+    "\n",
+    ">\\> MCHU - Calmodulin - Human, rabbit, bovine, rat, and chicken\n",
+    "MADQLTEEQIAEFKEAFSLFDKDGDGTITTKELGTVMRSLGQNPTEAELQDMINEVDADGNGTID\n",
+    "FPEFLTMMARKMKDTDSEEEIREAFRVFDKDGNGYISAAELRHVMTNLGEKLTDEEVDEMIREA\n",
+    "DIDGDGQVNYEEFVQMMTAK*\n",
+    "\n",
+    "We can see that in addition to the raw aminoacid sequence there is some metadata denoting where the sequence came from and what it represents.\n",
+    "\n",
+    "For our example, we will be using [CFTR gene](https://www.ensembl.org/Homo_sapiens/Gene/Summary?g=ENSG00000001626;r=7:117287120-117715971), encoding [a protein](https://www.uniprot.org/uniprot/P13569).\n",
+    "\n",
+    "First, let's download ther relevant FASTA file. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 147,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Downloaded file to:  data/ENSG00000001626.fasta\n",
+      "\n",
+      "Download metadata and statistics:\n",
+      "Vary: Content-Type\n",
+      "Vary: Origin\n",
+      "Content-Type: text/x-fasta; charset=UTF-8\n",
+      "Date: Tue, 23 Nov 2021 18:45:51 GMT\n",
+      "X-RateLimit-Limit: 55000\n",
+      "X-RateLimit-Reset: 849\n",
+      "X-Runtime: 1.262950\n",
+      "Connection: close\n",
+      "X-RateLimit-Period: 3600\n",
+      "X-RateLimit-Remaining: 54998\n",
+      "Content-Length: 436062\n",
+      "\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "from urllib.request import urlretrieve \n",
+    "\n",
+    "CFTR_FASTA_path = \"data/ENSG00000001626.fasta\"\n",
+    "result_location, http_response = urlretrieve(\"https://rest.ensembl.org/sequence/id/ENSG00000001626.fasta\", CFTR_FASTA_path)\n",
+    "print(\"Downloaded file to: \", result_location)\n",
+    "\n",
+    "print(\"\\nDownload metadata and statistics:\")\n",
+    "print(http_response)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "editable": false
+   },
+   "source": [
+    "The above message may look complicated - it contains a lot of information related to the transfer and the type of the file. \n",
+    "\n",
+    "To open the file and read relevant information from it, we use `SeqIO` module."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Reading FASTA files. Note that each file can contain multiple records.\n",
+    "from Bio import SeqIO\n",
+    "records = [r for r in SeqIO.parse(CFTR_FASTA_path, \"fasta\")]\n",
```

Comment:
Again, consider expanding the list creation

Reply:
Adjusted.